### PR TITLE
DRILL-5415: Improve Fixture Builder to configure client properties an…

### PR DESCRIPTION
**Pull request by sohami**
…d keep collection type properties for server

### CODE REVIEW

File: exec/java-exec/src/test/java/org/apache/drill/test/ClusterFixture.java [hunk 1]
```diff
@@ -203,7 +219,15 @@ private void createConfig(FixtureBuilder builder) throws Exception {
     if (builder.configResource != null) {
       config = DrillConfig.create(builder.configResource);
     } else if (builder.configProps != null) {
-      config = DrillConfig.create(configProperties(builder.configProps));
+      // Create Config object out of all the String type configuration.
```
Comment (paul-rogers) [inline exec/java-exec/src/test/java/org/apache/drill/test/ClusterFixture.java]: Suggestion to keep things tidy. Keep all properties (simple and complex) in the builder's `configProps` object. Then, in `configProperties`, do the mapping as required to create the `DrillConfig`. If a value is a collection, use the new code to set the value. Else, convert the value using `toString()`.

File: exec/java-exec/src/test/java/org/apache/drill/test/FixtureBuilder.java [hunk 2]
```diff
@@ -117,7 +122,27 @@ public FixtureBuilder configProperty(String key, Object value) {
     if (configProps == null) {
       configProps = defaultProps();
     }
-    configProps.put(key, value.toString());
+
+    if(value instanceof Collection<?>) {
```
Comment (paul-rogers) [inline exec/java-exec/src/test/java/org/apache/drill/test/FixtureBuilder.java]: Can we just leave the collection in the single `configProps` object, and do the mapping when creating the final config? That is, for the DrillConfig, make the collection test when building DrillConfig. See comment above.

File: exec/java-exec/src/test/java/org/apache/drill/test/ClusterFixture.java (hunk 1 above)
Comment (sohami) [inline exec/java-exec/src/test/java/org/apache/drill/test/ClusterFixture.java]: Fixed

File: exec/java-exec/src/test/java/org/apache/drill/test/FixtureBuilder.java (hunk 2 above)
Comment (sohami) [inline exec/java-exec/src/test/java/org/apache/drill/test/FixtureBuilder.java]: Fixed

**Comment by amansinha100** [comment]
+1

**Pull request closed without merging**